clients/src/main/java/org/apache/kafka/common/requests/JoinGroupRequest.java

```diff
@@ -189,6 +209,13 @@ public JoinGroupRequest(Struct struct, short versionId) {
             rebalanceTimeout = sessionTimeout;
 
         memberId = struct.get(MEMBER_ID);
+        // Only v5 and above APIs have group.instance.id field.
+        if (struct.hasField(GROUP_INSTANCE_ID)) {
```

Comment:
So clearly the intent is to silently fall back to the old join group logic, which means we become a dynamic member. It may be helpful having a log message indicating that this has happened.

One additional note: if the brokers are later upgraded to a version that does support static membership, we don't have any logic to detect it. I think this is probably fine, just worth keeping in mind.

Reply:
sure thing.